### PR TITLE
task 6 complete and ready for review

### PR DESCRIPTION
**Pull request by jxnbox**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -98,19 +98,68 @@ describe('3. Get api/articles/:article_id', () => {
...
+                            body : expect.any(String),
+                            created_at : expect.any(String),
+                            votes : expect.any(Number)
+                        })
+                    )
+                })
+            })
+        })
+    })
+
```
Comment (B7Bprog) [inline __tests__/app.test.js:158]: Please write a test to check if comments are served starting with the most recent one. (U can use jest-sorted).

Comment (B7Bprog) [inline __tests__/app.test.js:158]: Please add another test where the article ID is valid, but there are no comments for that article. It should respond with status 200 and an empty array.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -98,19 +98,68 @@ describe('3. Get api/articles/:article_id', () => {
...
+    describe('b. error handling', () => {
+
+        it('send a 400 status code when user inputs a bad request', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Please change the test description here as a 'bad request' could come in many different shapes. ( something like 'article ID is the wrong datatype' would be more descriptive to say.)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -98,19 +98,68 @@ describe('3. Get api/articles/:article_id', () => {
...
+
+describe('4. GET /api/articles/:article_id/comments', () => {
+    describe('a. status 200 & data', () => {
+        it('returns 200 status code & an array back to the user where the data', () => {
+            return request(app)
+            .get('/api/articles/3/comments')
+            .expect(200)
+            .then((res) => {
+                const articles = res.body;
+                expect(articles).toBeInstanceOf(Array);
```
Comment (B7Bprog) [inline __tests__/app.test.js:131]: You are using .forEach() array method, so is it necessary to check for array here?

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Jan a few modifications needed , please see my comments, amend your code and resubmit the pull request.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -118,4 +118,54 @@ describe('3. Get api/articles/:article_id', () => {
...
+                expect(articles).toBeSorted({ descending: true })
+            })
+        })
+
+        it('returns 200 status code & empty array when a valid ID is passed but contains no comments', () => {
+            return request(app)
+                   .get('/api/articles/7/comments')    
+                   .expect(200)
+                   .then( (res) => {
+                        const articles = res.body;
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: just be careful with your variable naming - the data on this endpoint is an array of `comments` but you've named the variable `articles`

File: __tests__/app.test.js [hunk 5]
```diff
@@ -118,4 +118,54 @@ describe('3. Get api/articles/:article_id', () => {
...
+        it('send a 400 status code when user inputs the wrong datatype for the ID', () => {
+            return request(app)
+                .get('/api/articles/article3/comments')
+                .expect(400)
+                .then ((res) => {
+                    const {msg} = res.body;
+                    expect(msg).toBe('Bad Request')
+                })
+        })
+    })
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:180]: just missing one more test for the `404` when a user inputs an `article_id` which is not found in the database, for example `999` or `0`

File: controllers.js [hunk 6]
```diff
@@ -23,9 +24,19 @@ exports.getArticle = (req, res, next) => {
...
     })
     .catch(next);
+}
+
+exports.getCommentsById = (req, res, next) => {
+    const article_id = req.params.article_id;
+
+    getCommentsByIdModel(article_id)
+    .then( (articles) => {
+        res.status(200).send(articles);
```
Comment (Pipe-Smoking-Rabbit) [inline controllers.js]: when you implement the test for the `404` error of an article_id of `999` or any number that is not found - you will find that it fails because you are not currently checking the database for if the article exists 

Here in the controller you can make use of another model function you have already written - `getArticleById` - and using a `Promise.all` or a promise chain of `.then` it can be used together with your `getCommentsByIdModel`

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Nice work on this one, it just needs another test for a `404` when the `article_id` is a number that cannot be found in the database. Check my comments for guidance and don't be afraid to nchelp if it feels unclear!

File: __tests__/app.test.js [hunk 7]
```diff
@@ -118,4 +118,64 @@ describe('3. Get api/articles/:article_id', () => {
...
+                        expect.objectContaining({
+                            article_id : 3,
+                            author : expect.any(String),
+                            body : expect.any(String),
+                            created_at : expect.any(String),
+                            votes : expect.any(Number)
+                        })
+                    )
+                })
+                expect(articles).toBeSorted({ descending: true })
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: can you specify what the articles are being sorted by? Have a look at .toBeSortedBy instead

File: __tests__/app.test.js [hunk 8]
```diff
@@ -118,4 +118,64 @@ describe('3. Get api/articles/:article_id', () => {
...
+            })
+        })
+
+        it('returns 200 status code & empty array when a valid ID is passed but contains no comments', () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:148]: nice edge case

**Review by roisinoreilly — APPROVED** [review]
nicely refactored, I've left a couple of comments but merge when ready
